File: platforms/android/devkit/devkit/src/main/java/io/nativeblocks/devkit/util/DevKitLogger.kt
package io.nativeblocks.devkit.util

import android.util.Log

internal object DevKitLogger {
    private const val isDebuggable = false
    fun d(tag: String?, msg: String) {
        if (isDebuggable)
            Log.d(tag, msg)
    }

    fun d(tag: String?, msg: String?, tr: Throwable?) {
        if (isDebuggable)
            Log.d(tag, msg, tr)
    }

    fun e(tag: String?, msg: String) {
        if (isDebuggable)
            Log.e(tag, msg)
    }

    fun e(tag: String?, msg: String?, tr: Throwable?) {
        if (isDebuggable)
            Log.e(tag, msg, tr)
    }
}

File: platforms/ios/devkit/Sources/NativeblocksDevkit/util/DevKitLogger.swift
import Foundation

internal struct DevKitLogger {
    private static let isDebuggable = false
    static func debug(_ tag: String, _ message: String) {
        if DevKitLogger.isDebuggable {
            print("DevKitLogger \(tag): \(message)")
        }
    }

    static func error(_ tag: String, _ message: String, _ error: Error) {
        if DevKitLogger.isDebuggable {
            print("DevKitLogger \(tag): \(message)", error)
        }
    }

    static func error(_ tag: String, _ error: Error) {
        if DevKitLogger.isDebuggable {
            print("DevKitLogger \(tag):", error)
        }
    }
}
